### ISSUE TITLE
TimeoutError on blocking indefinite wait

### ISSUE THREAD

**Issue by jakirkham** [ipython/ipyparallel#163]
Getting the following error with ipyparallel 5.1.1 on python 3.5. Basically using ipyparallel's `map` and iterating over the returned object for each result. This is a blocking (no timeout) wait on my end. So, it is pretty interesting to see `TimeoutError` produced.

```
---------------------------------------------------------------------------
TimeoutError                              Traceback (most recent call last)
/groups/dudman/home/kirkhamj/miniconda/envs/nanshenv3/lib/python3.5/site-packages/ipyparallel/client/asyncresult.py in _ordered_iter(self)
    694         try:
--> 695             rlist = self.get(0)
    696         except error.TimeoutError:

/groups/dudman/home/kirkhamj/miniconda/envs/nanshenv3/lib/python3.5/site-packages/ipyparallel/client/asyncresult.py in get(self, timeout)
    167         else:
--> 168             raise error.TimeoutError("Result not ready.")
    169 

TimeoutError: Result not ready.
```

**Comment by jakirkham** [comment]
Should add that this actually results in another error being thrown. See this issue ( https://github.com/ipython/ipyparallel/issues/162 ).

**Comment by jakirkham** [comment]
Also, should add that I don't encounter this error with 5.0.1.

**Comment by minrk** [comment]
Can you provide some samples? I can't seem to produce this with 5.1.1. I saw this in 5.1.0, but thought I fixed it in 5.1.1.

**Comment by jakirkham** [comment]
Basically [ditto]( https://github.com/ipython/ipyparallel/issues/162#issuecomment-231111476 ). 😄
